web-ts/src/components/LocationHeader.tsx

```diff
@@ -1,47 +1,63 @@
-import React, { ReactNode } from 'react'
-import { withTranslation, TFunction } from 'react-i18next'
+import React, { ReactNode, ReactElement } from 'react'
+import { generatePath } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
 import LanguageSelector from './LanguageSelector'
 import searchIcon from '../assets/magnifier.svg'
 import landingIcon from '../assets/location-icon.svg'
 import Header from './Header'
 import HeaderNavigationItem from '../components/HeaderNavigationItem'
-import { CityModel } from 'api-client'
+import {
+  CATEGORIES_ROUTE,
+  CityModel,
+  EVENTS_ROUTE,
+  LANDING_ROUTE,
+  LOCAL_NEWS_TYPE,
+  OFFERS_ROUTE,
+  POIS_ROUTE,
+  SEARCH_ROUTE,
+  TU_NEWS_TYPE
+} from 'api-client'
 import offersIcon from '../assets/offers.svg'
 import localInformationIcon from '../assets/local_information.svg'
 import eventsIcon from '../assets/events.svg'
 import newsIcon from '../assets/news.svg'
 import poisIcon from '../assets/pois.svg'
 import HeaderActionBarItemLink from '../components/HeaderActionItemLink'
 import buildConfig from '../constants/buildConfig'
+import { RoutePatterns } from '../routes/RootSwitcher'
 
 type PropsType = {
   cityModel: CityModel
   pathname: string
   languageCode: string
   viewportSmall: boolean
-  t: TFunction
   onStickyTopChanged: (stickyTop: number) => void
   languageChangePaths: Array<{ code: string; path: string | null; name: string }> | null
 }
 
-export class LocationHeader extends React.Component<PropsType> {
-  getActionItems(): Array<ReactNode> {
-    const { languageCode, pathname, languageChangePaths, t } = this.props
+const LocationHeader = (props: PropsType): ReactElement => {
+  const { viewportSmall, onStickyTopChanged, cityModel, languageCode, pathname, languageChangePaths } = props
+  const { eventsEnabled, poisEnabled, offersEnabled, tunewsEnabled, pushNotificationsEnabled } = cityModel
 
+  const params = { cityCode: cityModel.code, languageCode }
+  const categoriesPath = generatePath(RoutePatterns[CATEGORIES_ROUTE], params)
+  const eventsPath = generatePath(RoutePatterns[EVENTS_ROUTE], params)
+  const offersPath = generatePath(RoutePatterns[OFFERS_ROUTE], params)
+  const poisPath = generatePath(RoutePatterns[POIS_ROUTE], params)
+  const newsPath = generatePath(RoutePatterns[pushNotificationsEnabled ? LOCAL_NEWS_TYPE : TU_NEWS_TYPE], params)
```

Comment:
I think `generate` is not the right term here as it referes to randomness somehow.

Maybe just `create`.

Reply:
See answer above, `generatePath` is a function of `react-router`, see https://reactrouter.com/web/api/generatePath.